source/server/server.cc

```diff
@@ -517,9 +519,13 @@ void InstanceImpl::terminate() {
 Runtime::Loader& InstanceImpl::runtime() { return Runtime::LoaderSingleton::get(); }
 
 void InstanceImpl::shutdown() {
-  shutdown_ = true;
-  restarter_.terminateParent();
-  dispatcher_->exit();
+  dispatcher_->post([this] {
```

Comment:
I'm pretty sure this method must have always run on the main thread due to the `terminateParent()` call, right? Why is the post now necessary?

Reply:
Yup, you're right. shutdown is called from the admin [quitquitquit](https://github.com/envoyproxy/envoy/blob/84e67aa3ffbb5546a80c3d0c36e308e8dad1802f/source/server/http/admin.cc#L922) handler and I wasn't sure if that was always run on the main thread but integration tests which exercise this code path still pass without this change so I'll revert it.